src/components/Calculator.jsx

```diff
@@ -1,84 +1,93 @@
-/* eslint-disable indent */
 import React from 'react';
-
 import './calculator.css';
+import calculate from './calculator/logic/calculate';
+import Button from './buttons/buttons';
 
 export default class Calculator extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      calObj: { total: null, next: null, operation: null },
+      display: null,
+    };
+    this.handleClick = this.handleClick.bind(this);
+    this.buttons = [
+      'Ac',
+      '+/-',
+      '%',
+      '÷',
+      '7',
+      '8',
+      '9',
+      'x',
+      '4',
+      '5',
+      '6',
+      '-',
+      '1',
+      '2',
+      '3',
+      '+',
+      '0',
+      '.',
+      '=',
+    ];
+  }
+
+  handleDisplayState(beforeCalObj, afterCalObj) {
+    // handle display UI state
+    if (!afterCalObj.operation && !afterCalObj.next && afterCalObj.total) {
+      this.setState({ display: afterCalObj.total });
+    } else if (beforeCalObj.operation && afterCalObj.next) {
+      this.setState({ display: afterCalObj.next });
+    } else if (afterCalObj.total && afterCalObj.operation) {
+      this.setState({ display: afterCalObj.total });
+    } else if (!afterCalObj.operation) {
+      this.setState({ display: afterCalObj.next });
+    } else {
+      this.setState({ display: 0 });
+    }
+  }
+
+  handleClick(e) {
+    const { calObj } = this.state;
+    const newCalObj = calculate(calObj, e.target.innerText);
+    this.setState({ calObj: newCalObj });
+    this.handleDisplayState(calObj, newCalObj);
+  }
+
   render() {
+    const { display, calObj } = this.state;
     return (
       <div className="calculator">
-        <div className="display">0</div>
+        <div className="display">
+          {display ?? calObj.next ?? 0}
+        </div>
         <div className="keyboard">
           <div className="keyboard-row">
-            <button type="button" className="keyboardBtn">
-              AC
-            </button>
-            <button type="button" className="keyboardBtn">
-              +/-
-            </button>
-            <button type="button" className="keyboardBtn">
-              %
-            </button>
-            <button type="button" className="keyboardBtn">
-              &divide;
-            </button>
+            {this.buttons.slice(0, 4).map((btn) => (
+              <Button key={btn} name={btn} handleClick={this.handleClick} />
+            ))}
```

Comment:
- [ ] Please the `AC` button of your calculator is not working, please console message and kindly fix the issue.
![acbutton](https://user-images.githubusercontent.com/65852621/164359779-940fed5d-f9df-41cf-ba77-e2ed430ac9d5.png)